### PR TITLE
First run dialog aesthetic improvements

### PR DESCRIPTION
**Pull request by timlinux**
Updates the first run dialog that is shown to users migrating from QGIS Version 2.x to 3.x.

For users of development versions, they will be shown a message thanking them for using and testing the dev version.

first-run-dev-version

For users of release versions, they will be shown a message with a link to the current changelog.

first-run-prod-version

This completes work started in https://github.com/qgis/QGIS/pull/38616

### CODE REVIEW

**Comment by rduivenvoorde** [comment]
Nice, so this hides the 'import my ancient settings' IF the .qgis2 dir is just not there?

Does it take the current 'language' and 'version' into account for the visual changelog url? (Or is this handled by browser stuff??? I do not think so). Not sure if it is worth the effort, that dialog you do not see very often?

File: src/app/qgsfirstrundialog.cpp [hunk 1]
```diff
@@ -25,14 +25,25 @@ QgsFirstRunDialog::QgsFirstRunDialog( QWidget *parent ) : QDialog( parent )
...
+  QString minor = versionParts.at( 1 );
+  if ( minor.toInt() % 2 == 1 )
+  {
+    // Development version doesn't show the link to the changelog
+    mWelcomeProdLabel->hide();
+  }
+  else
+  {
+    // Production version shows link.
+    mWelcomeProdLabel->setText( mWelcomeProdLabel->text().replace( QString( "VERSION_TOKEN" ), major.append( minor ) ) );
```
Comment (m-kuhn) [inline src/app/qgsfirstrundialog.cpp:39]:
```suggestion
    mWelcomeProdLabel->setText( mWelcomeProdLabel->text().replace( QStringLiteral( "VERSION_TOKEN" ), major.append( minor ) ) );
```

**Comment by m-kuhn** [comment]
Cool, so now we always see the source code in QtCreator when launching as in the screen casts? That is truly open source !! :sweat_smile: 

If there is no previous settings detected, should we still show a first run dialog with a warm welcome and a link to interesting resources (documentation, how to contribute)?

**Comment by timlinux** [comment]
> Cool, so now we always see the source code in QtCreator when launching as in the screen casts? That is truly open source !! sweat_smile

heeeee yes!

> 
> If there is no previous settings detected, should we still show a first run dialog with a warm welcome and a link to interesting resources (documentation, how to contribute)?

I think that would be a great addition! Let me come up with something....

**Comment by timlinux** [comment]
> Nice, so this hides the 'import my ancient settings' IF the .qgis2 dir is just not there?

Yes

> 
> Does it take the current 'language' and 'version' into account for the visual changelog url? (Or is this handled by browser stuff??? I do not think so). Not sure if it is worth the effort, that dialog you do not see very often?

It doesnt take any language stuff into account. I think that is going to be complicated since as far as I know there is not a 1:1 match between the translations on the web site and those on the Desktop QGIS app. I guess if you made 'empty' translations  on the web site for each locale we have on QGIS it could work.

> 
> Should/can we also link to a page or site for the dev users? I don't know, github issues, or https://qgis.org/en/site/getinvolved/index.html (language aware ;-) )

Cool idea, let me work on that.

**Comment by nyalldawson** [comment]
Manually marking as stale because stalebot has left for christmas break

**Comment by timlinux** [comment]
@nyalldawson can we merge this, I will address @rduivenvoorde 's suggestions in a separate PR

**Comment by nyalldawson** [comment]
Done

**Pull request merged**